python-checks/src/test/resources/checks/constantCondition.py

```diff
@@ -55,4 +55,85 @@ def ignored():
     pass
   # builtin constructors are out of scope
   if list():
-    pass
\ No newline at end of file
+    pass
+
+def variables(param):
+  if param:
+    x = 1
+  else:
+    x = 2
+
+  x = 0
+#     ^> {{Last assignment.}}
+  if x:  ... # Noncompliant
+#    ^
+
+  y = 42
+  if y: ... # Noncompliant
+
+def variable_not_constant(param):
+  x = param
+  if x: ... # OK
+
+  if param:
+    z = 1
+  else:
+    z = 2
+  if z: ... # OK
```

Comment:
I think it could be interesting, but in my opinion that would be more appropriate for  RSPEC-2589 and /or RSPEC-2583.
In this rule we only care about conditions being constant